PuzzleSlider/src/Engine/GameData.java

```diff
@@ -124,6 +137,23 @@ public GridPoint getEmptyTile() {
 		return mBlankSpot;
 	}
 	
+	public int getMoveCount() {
+		return mMoveCount;
+	}
+	
+	public int incrementMoveCount(){
+		mMoveCount++;
+		return mMoveCount;
```

Comment:
Why are you returning `mMoveCount` here? I do not see it is being used and you have getter for this protected variable.